syntax = "proto3";

package demo;

// 指定生成的 Go 代码的包名
option go_package = "proto/demo";

// 用户信息
message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  enum Gender {
    UNKNOWN = 0;
    MALE = 1;
    FEMALE = 2;
  }
  Gender gender = 4;
  repeated string phone_numbers = 5;
  
  // 嵌套消息
  message Address {
    string street = 1;
    string city = 2;
    string country = 3;
    string postal_code = 4;
  }
  repeated Address addresses = 6;
  
  // 时间戳
  int64 created_at = 7;
  bool is_active = 8;
}

// 请求消息
message GetUserRequest {
  int32 user_id = 1;
}

// 响应消息
message GetUserResponse {
  User user = 1;
  string message = 2;
  int32 status = 3;
}

// 服务定义
service UserService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
}